test/compilable/test18367.sh

```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env bash
 
-set -u -o pipefail
+set -u -o pipefail -e
 
 # dmd should not segfault on -X with libraries, but no source files
 out=$("$DMD" -conf= -X foo.a 2>&1)
-[ $? -eq 1 ] || exit 1
```

Comment:
That's exactly what `-e` is for. `set -euo pipefail` should probably be set by default, but that's a task for another PR.